Common/Tests/MockVsTests/MockClassifierAggregatorService.cs

```diff
@@ -42,32 +42,31 @@ public class MockClassifierAggregatorService : IClassifierAggregatorService {
 
         sealed class AggregatedClassifier : IClassifier, IDisposable {
             private readonly ITextBuffer _buffer;
-            private readonly List<IClassifier> _classifiers;
+            private readonly IReadOnlyList<IClassifier> _classifiers;
 
             public AggregatedClassifier(ITextBuffer textBuffer, IEnumerable<IClassifierProvider> providers) {
                 _buffer = textBuffer;
                 _classifiers = providers.Select(p => p.GetClassifier(_buffer)).ToList();
+                foreach (var c in _classifiers) {
+                    c.ClassificationChanged += Subclassification_Changed;
+                }
             }
 
-            public void Dispose() {
-                foreach (var c in _classifiers.OfType<IDisposable>()) {
-                    c.Dispose();
-                }
+            private void Subclassification_Changed(object sender, ClassificationChangedEventArgs e) {
+                var c = (IClassifier)sender;
+                var refreshSpans = c.GetClassificationSpans(e.ChangeSpan);
```

Comment:
>GetClassificationSpans [](start = 37, length = 22)

The result of this is thrown away. Is there a desired side effect that is achieved by calling this?

Reply:
It ensures the token cache is filled for PythonClassifier. The sequence of events is strange, and probably some assumptions based on how VS works need to be fixed, but it goes roughly like this:

* create new text view with text already in it
* classifier attaches and we get classifications to fill the cache (implicit in VS, already explicit elsewhere in our tests)
* analysis entry gets assigned to the buffer, which resets the cache and raises the event (happens in both)
* in VS, and with this call here, the event triggers getting the new classifications, which refills the cache
* we delete the text from the buffer, which deletes the relevant part of the cache

At the last step, if the cache was not refilled then we throw an exception. This exception never seems to occur in actual use, because the aggregator gets the new classifications immediately. However, our test previously was not getting classifications immediately.

The token cache uses "highly optimized" array operations that rely on it being in a certain state. I don't want to touch it just yet, so I set up the mock tests to do the same steps as VS does (which arguably is the right way to do it - if there are problems in the classifier's assumptions, we want them to show up in both VS and mocks).